File: androidApp/src/main/java/com/vendhan/kmmmovies/android/ui/movies/TopRatedMoviesContent.kt
package com.vendhan.kmmmovies.android.ui.movies

import androidx.compose.runtime.Composable
import com.vendhan.kmmmovies.repo.model.MoviesResult

@Composable
fun TopRatedMoviesContent(
    title: String,
    movies: List<MoviesResult>,
    onClick: (Int) -> Unit
) {
    MoviesHeader(
        title = title
    )
    movies.forEach { movie ->
        MoviesCardHorizontal(
            movie = movie,
            onClick = onClick
        )
    }
}

File: iosApp/iosApp/Movies/TopRatedMoviesContent.swift
//
//  TopRatedMoviesContent.swift
//  iosApp
//
//  Created by Arul Vendhan Arumugam on 02/09/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import SwiftUI
import shared

struct TopRatedMoviesContent: View {
    
    var title : String
    var movies : [MoviesResult] = []
    
    init(title : String, movies : [MoviesResult]){
        self.title = title
        self.movies = movies
    }
    
    var body: some View {
        MoviesHeader(title: title)
        ForEach(movies, id: \.id) { movie in
            NavigationLink(destination: DetailsScreen(movieID: Int(movie.id))) {
                MoviesCardHorizontal(movie: movie)
            }
            .buttonStyle(PlainButtonStyle())
        }
    }
}
